### PR TITLE
Add VQSelect and HeapSelect

### PR DESCRIPTION
**Pull request by enum-class**
Implemented VQSelect by @zbjornson in #1710.
I've incorporated his patch with some refinements.
Also, added HeapSelect functionality.

### CODE REVIEW

File: hwy/contrib/sort/sort_test.cc [hunk 1]
```diff
@@ -732,6 +733,96 @@ void TestAllSort() {
   }
 }
 
+std::vector<Algo> SelectAlgoForTest() {
+  return {
+#if VQSORT_ENABLED
+//    Algo::kVQSelect,
```
Comment (jan-wassenberg) [inline hwy/contrib/sort/sort_test.cc:739]: Should we uncomment this?

File: hwy/contrib/sort/vqsort-inl.h [hunk 2]
```diff
@@ -42,6 +42,12 @@
 #define VQSORT_PRINT 0
 #endif
 
+#if HWY_CXX_LANG >= 201703L
```
Comment (jan-wassenberg) [inline hwy/contrib/sort/vqsort-inl.h:45]: Let's move this to base.h, for example near HWY_HAVE_CXX20_THREE_WAY_COMPARE.

**Review by jan-wassenberg — CHANGES_REQUESTED** [review]
Super, thanks for updating and improving this patch! Two minor suggestions, then we're good to merge :) (The MSVC fix is in.)

**Review by jan-wassenberg — APPROVED** [review]
Nice, thanks for making the changes.

**Comment by enum-class** [comment]
@jan-wassenberg Thanks for your comments, I have fixed issues.

**Pull request merged**